src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlBuffer.cs

```diff
@@ -856,7 +855,7 @@ internal object Value
                     case StorageType.Empty:
                         return DBNull.Value;
                     case StorageType.Boolean:
-                        return Boolean;
+                        return Boolean ? s_cachedTrueObject : s_cachedFalseObject;
```

Comment:
Comment for other reviewers: This change had me wondering if there might be hidden consequences here. But I see this is what the code is in netcore. So I assume it's been hashed out and tested.

Reply:
It's a change in added in corefx. Returning a literal boolean each time caused the value to be boxed into the object return variable location so if you were reading bool fields you got a lot of gen0 garbage. This removed the garbage by returning a pre-boxed boolean, essentially caching the value.

Reply:
A thought, Is it worth putting a comment in the code explaining this? Corefx tended to be very sceptical of comments but I think it's helpful to explain non-obvious things like this.

Reply:
Would appreciate an inline comment over the 2 private cached instances. Thanks for explanation!

Reply:
Done.